packages/graphql/src/registry.ts

```diff
@@ -39,61 +50,148 @@ interface TSPTypeContext {
  *    by using thunks for fields/arguments.
  */
 export class GraphQLTypeRegistry {
-  // Stores intermediate TSP type information, keyed by TSP type name.
-  // TODO: make this more of a seen set
-  private TSPTypeContextRegistry: Map<string, TSPTypeContext> = new Map();
+  // Global registry to prevent GraphQL type name collisions
+  static #globalNameRegistry = new Set<TypeKey>();
 
-  // Stores materialized GraphQL types, keyed by their GraphQL name.
-  private materializedGraphQLTypes: Map<string, GraphQLNamedType> = new Map();
+  // Type maps for different GraphQL types
+  #objectTypes: ObjectTypeMap;
 
-  addEnum(tspEnum: Enum): void {
-    const enumName = tspEnum.name;
-    if (this.TSPTypeContextRegistry.has(enumName)) {
-      // Optionally, log a warning or update if new information is more complete.
-      return;
-    }
+  // Program reference for using TypeSpec utilities
+  #program: Program;
+
+  // TypeSpec typekit for easy access to TypeSpec utilities
+  #typekit: ReturnType<typeof $>;
+
+  constructor(program: Program) {
+    // Initialize type maps with necessary dependencies
+    this.#objectTypes = new ObjectTypeMap();
+    this.#program = program;
+    this.#typekit = $(program);
+  }
 
-    this.TSPTypeContextRegistry.set(enumName, {
-      tspType: tspEnum,
-      name: enumName,
-      // TODO: Populate usageFlags based on TSP context and other decorator context.
-    });
+  /**
+   * Reset the global name registry 
+   */
+  static resetGlobalRegistry(): void {
+    GraphQLTypeRegistry.#globalNameRegistry.clear();
   }
 
-  // Materializes a TSP Enum into a GraphQLEnumType.
-  materializeEnum(enumName: string): GraphQLEnumType | undefined {
-    // Check if the GraphQL type is already materialized.
-    if (this.materializedGraphQLTypes.has(enumName)) {
-      return this.materializedGraphQLTypes.get(enumName) as GraphQLEnumType;
+  /**
+   * Get GraphQL type names for a model based on usage flags
+   * Returns a mapping of usage flags to their corresponding GraphQL type names
+   */
+  #getModelTypeNames(modelName: string): Record<UsageFlags, string | undefined> {
+    // For now, we only support output types
+    // TODO: Add support for input types when InputTypeMap is implemented
+    const outputTypeName = this.#objectTypes.isRegistered(modelName) ? modelName : undefined;
+
+    return {
+      [UsageFlags.None]: undefined,
+      [UsageFlags.Input]: undefined, // TODO: Implement when InputTypeMap is added
+      [UsageFlags.Output]: outputTypeName,
+    };
+  }
+
+  /**
+   * Register a TSP Model
+   */
+  addModel(model: Model): void {
+    const model_context: TSPContext<Model> = {
```

Comment:
I think the expected style is to use `pascalCase`.
Python habits are hard to break 🙂